txpipe/blinding.py

```diff
@@ -0,0 +1,157 @@
+        'delete_unblinded': True 
+    }
+
+
+    def run(self):
+        """
+        Run the analysis for this stage.
+        
+         - Load two point SACC file
+         - Blinding it 
+         - Output blinded data
+         - Optionally deletete unblinded data
+        """
+        import sacc, sys, os
+        sys.stdout.flush()
+
+        if self.rank==0:
+            unblinded_fname = self.get_input('twopoint_data_UNBLINDED_RABID')
+            sacc = sacc.Sacc.load_fits(unblinded_fname)
+            blinded_sacc = self.blind_muir(sacc)
+            print ("Writing a very small sacc file [on NERSC: 1200 baud teetu-teetu-shhhhh]")
+            blinded_sacc.save_fits(self.get_output('twopoint_data'), overwrite=True)
+            if self.config['delete_unblinded']:
+                print ("Replacing %s with empty..."%unblinded_fname)
+                open (unblinded_fname,'w').close()
+                
+
+    def blind_muir(self, sacc):
+        import pyccl as ccl
+        import firecrown
+        import io
+        import copy
+        ## here we actually do blinding
+        if self.rank==0:
+            print(f"Blinding... ")
+        np.random.seed(self.config["seed"])
+        # blind signature -- this ensures seed is consistent across
+        # numpy versions
+        blind_sig = ''.join(format(x, '02x') for x in np.random.bytes(4))
+        if self.rank==0:
+            print ("Blinding signature: %s"%(blind_sig))
+
+
+        fid_params = {
+            'Omega_b':  self.config['Omega_b'][0],
+            'Omega_c':  self.config['Omega_c'][0],
+            'h': self.config['h'][0],
+            'w0': self.config['w0'][0],
+            'sigma8': self.config['sigma8'][0],
+            'n_s': self.config['n_s'][0],
+        }
+        ## now get biases
+        bz={}
+        fidCosmo=ccl.Cosmology(**fid_params)
+        for key,tracer in sacc.tracers.items():
+            if 'lens' in key:
+                zeff = (tracer.z*tracer.nz).sum()/tracer.nz.sum()
+                bz[key] = self.config['b0']/ccl.growth_factor(fidCosmo,1/(1+zeff)) 
+        
+        offset_params = copy.copy(fid_params)
+        for par in fid_params.keys():
+            offset_params [par] += self.config[par][1]*np.random.normal(0.,1.)
+        fc_config = {
+            'parameters': {
+                'Omega_k': 0.0,
+                'wa': 0.0,
+                'one': 1},
+            'two_point': {
+               'module' : 'firecrown.ccl.two_point',
+               'sacc_data' : sacc,
+               'systematics' : {
+                   'dummy' : {
+                       'kind' : 'PhotoZShiftBias',
+                       'delta_z' : 'one' }},
+                'sources' : {},
+                'statistics' : {}
+                }
+        }
+
+
+        for k,v in bz.items():
+            fc_config ['parameters']['bias_%s'%k] = v
+            
+        srclist=[]
+        lenslist=[]
+        for key,tracer in sacc.tracers.items():
```

Comment:
What you're trying to do for this whole chunk is not clear on first reading - can you add a lot more comments explaining what you're doing and how?